File: ai-job-portal-frontend/src/features/auth/services/auth.service.ts
import { apiClient } from "@/services/api-client";
import type { ApiResponse } from "@/types/api";
import type { AuthResponse, UserResponse } from "@/types/auth";
import type {
  ChangePasswordRequest,
  ForgotPasswordRequest,
  GoogleAuthRequest,
  LoginRequest,
  RefreshTokenRequest,
  RegisterRequest,
  ResendVerificationRequest,
  ResetPasswordRequest,
  VerifyEmailRequest,
} from "@/features/auth/types";

/**
 * Maps 1:1 to auth-service's AuthController endpoints
 * (CommonConstants.API_BASE_PATH + "/auth"). No endpoint, request, or
 * response shape may be altered from the frontend, per
 * 04_BACKEND_INTEGRATION.md.
 */
export const authService = {
  register: (payload: RegisterRequest) =>
    apiClient
      .post<ApiResponse<UserResponse>>("/auth/register", payload)
      .then((res) => res.data),

  login: (payload: LoginRequest) =>
    apiClient.post<ApiResponse<AuthResponse>>("/auth/login", payload).then((res) => res.data),

  loginWithGoogle: (payload: GoogleAuthRequest) =>
    apiClient.post<ApiResponse<AuthResponse>>("/auth/oauth/google", payload).then((res) => res.data),

  refreshToken: (payload: RefreshTokenRequest) =>
    apiClient
      .post<ApiResponse<AuthResponse>>("/auth/refresh-token", payload)
      .then((res) => res.data),

  logout: (payload: RefreshTokenRequest) =>
    apiClient.post<ApiResponse<null>>("/auth/logout", payload).then((res) => res.data),

  verifyEmail: (payload: VerifyEmailRequest) =>
    apiClient.post<ApiResponse<null>>("/auth/verify-email", payload).then((res) => res.data),

  resendVerification: (payload: ResendVerificationRequest) =>
    apiClient
      .post<ApiResponse<null>>("/auth/resend-verification", payload)
      .then((res) => res.data),

  forgotPassword: (payload: ForgotPasswordRequest) =>
    apiClient.post<ApiResponse<null>>("/auth/forgot-password", payload).then((res) => res.data),

  resetPassword: (payload: ResetPasswordRequest) =>
    apiClient.post<ApiResponse<null>>("/auth/reset-password", payload).then((res) => res.data),

  changePassword: (payload: ChangePasswordRequest) =>
    apiClient.post<ApiResponse<null>>("/auth/change-password", payload).then((res) => res.data),

  getCurrentUser: () =>
    apiClient.get<ApiResponse<UserResponse>>("/auth/me").then((res) => res.data),
};

File: ai-job-portal-backend/auth-service/src/main/java/com/prahlad/aijobportal/authservice/auth/service/AuthService.java
package com.prahlad.aijobportal.authservice.auth.service;

import com.prahlad.aijobportal.authservice.auth.dto.request.ChangePasswordRequest;
import com.prahlad.aijobportal.authservice.auth.dto.request.ForgotPasswordRequest;
import com.prahlad.aijobportal.authservice.auth.dto.request.GoogleAuthRequest;
import com.prahlad.aijobportal.authservice.auth.dto.request.LoginRequest;
import com.prahlad.aijobportal.authservice.auth.dto.request.RefreshTokenRequest;
import com.prahlad.aijobportal.authservice.auth.dto.request.RegisterRequest;
import com.prahlad.aijobportal.authservice.auth.dto.request.ResendVerificationRequest;
import com.prahlad.aijobportal.authservice.auth.dto.request.ResetPasswordRequest;
import com.prahlad.aijobportal.authservice.auth.dto.request.VerifyEmailRequest;
import com.prahlad.aijobportal.authservice.auth.dto.response.AuthResponse;
import com.prahlad.aijobportal.authservice.auth.dto.response.UserResponse;

import java.util.UUID;

public interface AuthService {

    UserResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    /** DAY12 "Google OAuth": verifies the ID token, then logs in an existing account or auto-registers a new one. */
    AuthResponse loginWithGoogle(GoogleAuthRequest request);

    AuthResponse refreshToken(RefreshTokenRequest request);

    void logout(RefreshTokenRequest request);

    void verifyEmail(VerifyEmailRequest request);

    void resendVerificationEmail(ResendVerificationRequest request);

    void forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);

    void changePassword(UUID userId, ChangePasswordRequest request);

    UserResponse getCurrentUser(UUID userId);
}
